packages/cron-job/Tests/Command/QueueDueCronJobsCommandTest.php

```diff
@@ -0,0 +1,181 @@
+use Draw\Component\Tester\MockTrait;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Console\Command\Command;
+
+#[CoversClass(QueueDueCronJobsCommand::class)]
+class QueueDueCronJobsCommandTest extends TestCase
+{
+    use CommandTestTrait;
+    use MockTrait;
+
+    private ManagerRegistry&MockObject $managerRegistry;
+
+    private CronJobProcessor&MockObject $cronJobProcessor;
+
+    public function createCommand(): Command
+    {
+        return new QueueDueCronJobsCommand(
+            $this->managerRegistry = $this->createMock(ManagerRegistry::class),
+            $this->cronJobProcessor = $this->createMock(CronJobProcessor::class)
+        );
+    }
+
+    public function getCommandName(): string
+    {
+        return 'draw:cron-job:queue-due';
+    }
+
+    public static function provideTestArgument(): iterable
+    {
+        return [];
+    }
+
+    public static function provideTestOption(): iterable
+    {
+        return [];
+    }
+
+    /**
+     * @param array{name: string, due: bool}[] $rawCronJobs
+     * @param string[]                         $expectedDisplay
+     */
+    #[DataProvider('provideDataForTestExecute')]
+    public function testExecute(array $rawCronJobs, array $expectedDisplay): void
+    {
+        $this->managerRegistry
+            ->expects(static::any())
+            ->method('getRepository')
+            ->with(CronJob::class)
+            ->willReturn($repository = $this->createMock(EntityRepository::class));
+
+        $repository
+            ->expects(static::once())
+            ->method('findBy')
+            ->with(['active' => true])
+            ->willReturn(
+                $cronJobs = array_map(
+                    fn (array $rawCronJob): CronJob&MockObject => $this->createCronJob(
+                        $rawCronJob['name'],
+                        $rawCronJob['due']
+                    ),
+                    $rawCronJobs
+                )
+            );
+
+        $dueCronJobs = array_filter(
+            $cronJobs,
+            static fn (CronJob $cronJob): bool => $cronJob->isDue()
+        );
+
+        if (0 === $numDueCronJobs = \count($dueCronJobs)) {
+            $this->cronJobProcessor
+                ->expects(static::never())
+                ->method('queue');
+        } else {
+            $this->cronJobProcessor
+                ->expects(static::exactly($numDueCronJobs))
+                ->method('queue')
+                ->with(
+                    ...static::withConsecutive(...array_map(
+                        static fn (CronJob $cronJob): array => [$cronJob],
+                        $dueCronJobs
+                    ))
+                );
+        }
+
+        $this
+            ->execute([])
+            ->test(
+                CommandDataTester::create(
+                    Command::SUCCESS,
```

Comment:
FYI if you pass null it will not validated output.

This can be usefull for the "errorDisplay" since the progress bar get displayed there and it's hard to test.

Do not do a progress bar like you did in the begining, it will almost always be call via a cron, not one will see it.